### DISCUSSION TITLE
Modification of the vegetation layer input with BD Topo data input and calculation of indicators (SVF, ...)

### DISCUSSION THREAD

**Discussion by atoulou** [orbisgis/geoclimate#1080] [Q&A]
Hi,

I would like to compute LCZs using Geoclimate with BD Topo input data, but I would like to replace the vegetation layer provided in BD TOPO with another vegetation dataset I have. 

I have several questions : 

### 1/ About the vegetation layer attributes

I checked the required input files for a use of Geoclimate with BDTopo v3 data here:https://github.com/orbisgis/geoclimate/tree/master/bdtopo/src/test/resources/org/orbisgis/geoclimate/bdtopo/v3/sample_12174 

There is only one vegetation-related layer: ZONE_DE_VEGETATION
Thus, my idea is to replace the ZONE_DE_VEGETATION shapefile (i.e. .shp, .dbf, .shx, .prj) from BD TOPO with my own vegetation dataset.

I compared:
- The attributes of BD TOPO’s ZONE_DE_VEGETATION layer
- The vegetation specifications described in the geoclimate documentation: https://geoclimate.readthedocs.io/en/latest/Vegetation-layer.html

From this comparison, it seems that BD TOPO only provides the attribute "Nature", which corresponds to the "type" attribute in the GeoClimate documentation. The other attributes listed in the documentation are not present in BD TOPO.

Does this mean that those additional attributes are not required for the computations performed by GeoClimate?

**In other words, which attributes of the vegetation layer are actually used by the LCZ algorithm, and therefore which attributes must be present in my vegetation layer when I replace the BD TOPO vegetation file?**

### 2/ About SVF (Sky View Factor) and vegetation

In the documentation, it appears that SVF is calculated considering buildings only. However, ignoring vegetation in SVF computation may affect LCZ classification, especially when using a more detailed vegetation dataset.

Since I want to use another vegetation layer (different from BD TOPO), I would like this vegetation to influence:
-the surface cover indicators, and
-the SVF calculation.

I know that SVF can be computed externally, for example using the UMEP plugin in QGIS.
Therefore, **is it possible to manually replace the SVF layer used in the LCZ computation in GeoClimate?**
 
Thank you very much in advance for your help.

Best Regards

**Comment by j3r3m1** [comment]
Hi @atoulou

If I understand well, you would like to replace the vegetation data in the BDTOPO by your own and then run the BDTOPO GeoClimate workflow. It might work but would need probably more work than you imagine.

From a quick look at the code, here is what is done to create the GeoClimate vegetation data from the BDTOPO ones:
1. Items from three datasources are used: "zone de vegetation", "aerodrome" and "terrain de sport": [here is the code](https://github.com/orbisgis/geoclimate/blob/4c058ad5aabee54678868eb753dfc42c7e71f4ed/bdtopo/src/main/groovy/org/orbisgis/geoclimate/bdtopo/InputDataLoading.groovy#L621) showing what is selected
2. Then the vegetation types and vegetation height are set according to a set of possible values. The conversion is defined [in this code](https://github.com/orbisgis/geoclimate/blob/4c058ad5aabee54678868eb753dfc42c7e71f4ed/bdtopo/src/main/groovy/org/orbisgis/geoclimate/bdtopo/InputDataFormatting.groovy#L799)

We would rather recommend you to format the initial BDTOPO data into the GeoClimate inputs, then replace the vegetation data generated by GeoClimate from the BDTOPO by your data and last run the GeoClimate algorithm you like.

This means to use a Groovy script to call each specific functions of GeoClimate instead of the one that run the whole processing chain.

Would you mind sharing a sample of your vegetation data (let's say 500 m x 500 m) ? Then I can show you an example of script that works.

**Reply by atoulou** [reply]
Hi,

Thank you very much for your quick and detailed response, as well as for the associated code.

Here is a sample of my vegetation dataset.
I have been working with a raster file that I converted into a vector file.
The column "Strate_veg" contains values from 1 to 5, representing the following categories:
1 = grass / 2 = bushes (<1.5 m) / 3 = shrubs (1.5 m–5 m) / 4 = small trees (5 m–15 m) / 5 = tall trees (>15 m).

Thank you in advance for your help.

Best regards,

[veg_AT_sample.zip](https://github.com/user-attachments/files/23495220/veg_AT_sample.zip)

**Reply by j3r3m1** [reply]
Thanks for sharing. However, for now GeoClimate does not deal with bush and shrubs categories (ashame I know, planned but not yet achieved). If you would like to keep the degree of accuracy of your vegetation data, maybe the best solution is run GeoClimate without your vegetation dataset and then use your it only to update the GeoClimate LCZ which are land cover types (LCZA-G) ?

If you prefer we use it for segmentation and calculations in the GeoClimate algo, please provide an update sample with only two possibilities for the strate_veg (low and high).

**Reply by atoulou** [reply]
Thank you for the explanations and advice. Since I would like to explore the differences when using my own vegetation data, here is my updated sample that includes a new column called "height" with only 2 possible values : low and high.

**Reply by j3r3m1** [reply]
I cannot find your updated sample

**Reply by atoulou** [reply]
I must have forgotten to put it in, sorry about that. Here it is
[veg_AT_sample.zip](https://github.com/user-attachments/files/23577155/veg_AT_sample.zip)

**Reply by j3r3m1** [reply]
OK here are the steps to follow:
1. Run GeoClimate with BDTopo or OSM --> files are saved in "outputGeoClimate"
2. Create a copy of "outputGeoClimate" and call it "outputGeoClimateWithMyVeg". You will now work with this copy.
3. Remove the vegetation file in this folder, you will use your
4. Depending the RSU you want to use, several options:
  - if you want to use the RSU created by GeoClimate using the BDTopo data, keep the file "rsu_indicators" as it is.
  - if you want that the algorithm create new RSU according to the new vegetation file, remove the file "rsu_indicators".
  - if you want to use your own RSU (buffer around your stations), replace the "rsu_indicators" file by your file (same file name and the file should contain the geometry field called "THE_GEOM" and the is called "ID_RSU"). Please do not use RSU of small size (< 500 m radius) if your RSU boundaries intersect buildings.
6. Modify your vegetation layer in order to make it have the needed GeoClimate input format (https://geoclimate.readthedocs.io/en/latest/Vegetation-layer.html)  --> if you use Python, you can install h2gis and then use the Python scripts enclosed (make sure the script saves the vegetation in the "outputGeoClimateWithMyVeg" folder)
7. Run the groovy script as follow in a terminal: groovy /path/to/groovyScript.groovy /path/to/outputGeoClimateWithMyVeg /path/to/outputDir RSU

[Scripts.zip](https://github.com/user-attachments/files/23959908/Scripts.zip)

**Reply by j3r3m1** [reply]
Could you please give us more about the use you will have from GeoClimate ?

**Reply by atoulou** [reply]
Thank you very much for the detailed workflow and the scripts you shared. I will try using them.

I would like to use GeoClimate as part of my doctoral research. More specifically, I am seeking to explore the relationships between local climatic conditions (approximated by LCZs, with all the limitations that this entails) and biodiversity data. I therefore intend to use GeoClimate to produce LCZ data for my study area.

**Comment by j3r3m1** [comment]
Concerning your second question, indeed the SVF in GeoClimate does not account for vegetation. But I do not think it is relevant to change the way the SVF is calculated, below is described why.

Currently, the GeoClimate LCZ algorithm differs whether a spatial unit is urban (building density > 0.1 and H/W > 0.1) or rural. If urban, only built-up parameters are considered. Only the morphology and building types due to buildings is considered to set a LCZ type. Thus the amount of vegetation or the SVF due to vegetation should come as a second indicator, complementary of the first one which is the LCZ type.

**Reply by atoulou** [reply]
Thank you again for your answer and explanations. 
However, I do not understand why, for the LCZ built-types (LCZ 1 to 10), the vegetation would be only a second indicator after the purely built indicators ? Is it a choice made in the Geoclimate algorithm ? And if it is, why was [this](url) choice made? 
Moreover, in the case of vegetation's impact on SVF it is not even a secondary indicator as it is never taken into account, isn't it ? 

Thank you again for your explanations.

**Reply by j3r3m1** [reply]
Using only building related indicators to determine the LCZ type of a built-type is a choice made in the GeoClimate algorithm. For us the names of the LCZ is important. If I have a LCZ 2 (called compact mid-rise) with no vegetation at all, it cannot become a LCZ 5 (open mid-rise) only because I have added vegetation in the streets. Thus, I do not say that the vegetation indicators are used as secondary indicators to make a decision on the LCZ type, I say that they are not used at all. However, I say that the amount of vegetation can still be a useful information to differentiate a LCZ 2 with few or a lot of trees. In this sense it is a secondary indicator since building morphology predominates. 

Concerning LCZ land-cover types, the SVF could be an indicator useful to differentiate the land cover but I do not consider it is fundamental anyway at the LCZ scale. However, it is much more useful at finer resolution, when you are interested in where exactly will be the shadow.

**Reply by atoulou** [reply]
Ok I understand your point, thank you, is this explained in a paper ?

And just to be sure, unlike the SVF, vegetation is taken into account through the pervious surface fraction when determining the LCZ type, right ?

**Reply by j3r3m1** [reply]
> Ok I understand your point, thank you, is this explained in a paper ?

The whole LCZ algorithm is described in this paper: https://doi.org/10.5194/gmd-17-2077-2024

> And just to be sure, unlike the SVF, vegetation is taken into account through the pervious surface fraction when determining the LCZ type, right ?

For LCZ built types, pervious fraction is currently not even considered. But it is very simple to consider it, a single input has to be modified in order to consider it (but we do not recommend).
For LCZ land-cover types the fraction of low and high vegetation are considered, along with other fractions (see the Figure  2 of the paper described above)

**Reply by atoulou** [reply]
Ok thank you for these clarifications

**Reply by atoulou** [reply]
I went through the article you provided in detail, and I’m sorry to come back to the topic again, but some points are still not entirely clear to me.

The first point concerns the choice made in GeoClimate to determine the built LCZ classes (LCZ 1 to 9, except LCZ 8) solely from built-related indicators, without integrating vegetation indicators.
I understand the argument that built LCZ classes are, by definition, named according to built characteristics. However, in the original LCZ standard, vegetation-related indicators are indeed used to characterize the classes — and, as far as I can tell, the standard does not state that they should be excluded when determining LCZ types.
This is why I have difficulty understanding the methodological choice implemented in the GeoClimate algorithm.
Is the idea that built-related indicators might indirectly capture the information contained in vegetation indicators?
In other words: if a spatial unit falls perfectly within the standard ranges for all built indicators of a given LCZ (for example LCZ 1), would it necessarily — or at least mostly — fall within the standard ranges of the vegetation indicators as well?

The second point relates to a test I had conducted before asking my questions here.
At that time, I had replaced the BDTopo vegetation data with my own dataset, assigning all features to the “hedge” type. I am fully aware that I had not properly formatted my data, so the results were necessarily incorrect.
However, I observed that some built LCZ classes changed after this modification (for example, an LCZ 5 becoming an LCZ 2).
If vegetation-based indicators are not used to determine built LCZ classes, I do not understand how such changes could occur.
Could this be due to modifications in the definition of the RSU (for instance, if vegetation patches previously ignored become included because their area exceeds the 10,000 m² threshold)?
Or is there another explanation?

Thank you again for your help.

**Reply by j3r3m1** [reply]
> I went through the article you provided in detail, and I’m sorry to come back to the topic again, but some points are still not entirely clear to me.
> 
> The first point concerns the choice made in GeoClimate to determine the built LCZ classes (LCZ 1 to 9, except LCZ 8) solely from built-related indicators, without integrating vegetation indicators. I understand the argument that built LCZ classes are, by definition, named according to built characteristics. However, in the original LCZ standard, vegetation-related indicators are indeed used to characterize the classes — and, as far as I can tell, the standard does not state that they should be excluded when determining LCZ types. This is why I have difficulty understanding the methodological choice implemented in the GeoClimate algorithm. Is the idea that built-related indicators might indirectly capture the information contained in vegetation indicators? In other words: if a spatial unit falls perfectly within the standard ranges for all built indicators of a given LCZ (for example LCZ 1), would it necessarily — or at least mostly — fall within the standard ranges of the vegetation indicators as well?

I do not think so but you might test this assumption if you have accurate vegetation data. The fact that the vegetation was not taken into account in the algorithm is also explained by the fact that the datasources initially used in GeoClimate (BDTOPO and OSM) does not have very accurate vegetation data. In the groovy script I have provided you will be able to modify the weights and then see how much the results are affected. We have actually made quite a lot of tests when initially defining the algorithm and ended with the current ones. However, it is still quite subjective and you may for example investigate the weight values that fit the best with the training data provided by the WUDAPT project. If so we would be please to work with you on this topic.   

> The second point relates to a test I had conducted before asking my questions here. At that time, I had replaced the BDTopo vegetation data with my own dataset, assigning all features to the “hedge” type. I am fully aware that I had not properly formatted my data, so the results were necessarily incorrect. However, I observed that some built LCZ classes changed after this modification (for example, an LCZ 5 becoming an LCZ 2). If vegetation-based indicators are not used to determine built LCZ classes, I do not understand how such changes could occur. Could this be due to modifications in the definition of the RSU (for instance, if vegetation patches previously ignored become included because their area exceeds the 10,000 m² threshold)? Or is there another explanation?
> 
> Thank you again for your help.

You have modified the BDTOPO data directly I suppose ? If this is the case the only explanation I see is the one you provide. You should check that for the RSU that had their LCZ type changed what was the values of the indicators used for the classification. I suppose they have changed. Please feel free to share the results of your investigation.
